src/procedures/RemoveDocument.ts

```diff
@@ -0,0 +1,60 @@
+import { Procedure } from './Procedure';
+import {
+  ProcedureType,
+  PolyTransactionTag,
+  RemoveDocumentProcedureArgs,
+  ErrorCode,
+} from '../types';
+import { PolymathError } from '../PolymathError';
+
+export class RemoveDocument extends Procedure<RemoveDocumentProcedureArgs> {
+  public type = ProcedureType.RemoveDocument;
+
+  public async prepareTransactions() {
+    const { symbol, name } = this.args;
+    const { contractWrappers, currentWallet } = this.context;
+
+    let securityToken;
+
+    try {
+      securityToken = await contractWrappers.tokenFactory.getSecurityTokenInstanceFromTicker(
+        symbol
+      );
+    } catch (err) {
+      throw new PolymathError({
+        code: ErrorCode.ProcedureValidationError,
+        message: `There is no Security Token with symbol ${symbol}`,
+      });
+    }
+
+    const [owner, account] = await Promise.all([securityToken.owner(), currentWallet.address()]);
+
+    if (account !== owner) {
+      throw new PolymathError({
+        code: ErrorCode.ProcedureValidationError,
+        message: `You must be the owner of this Security Token to remove a document`,
+      });
+    }
+
+    if (name.length < 1 || name.length > 32) {
```

Comment:
Checking the contract wrappers and the contracts themselves, I found no reference to that 32 character limit. The only limitation I found was that the values had to be at least 1 char long